encoder.go

```diff
@@ -30,6 +30,14 @@ type native interface {
 	int64 | uint64 | uintptr
 }
 
+func newLimitedEncoder() *encoder {
+	return &encoder{
+		containerMaxSize: wafMaxContainerSize,
+		stringMaxSize:    wafMaxStringLength,
+		objectMaxDepth:   wafMaxContainerDepth,
+	}
+}
```

Comment:
Out of curiosity - why does this need to be a pointer? It feels to me like this could just be returning `encoder` and it wouldn't change a thing... Same goes for `newMaxEncoder`... There is no interface involved here (that I'm aware of)...